### PR TITLE
generator keep consistency

### PR DESCRIPTION
**Pull request by njuettner**
To keep consistency I would suggest to use len(leader).

url concat is maybe confusing (at least for me), instead it could be replaced by fmt.Sprintf function

### CODE REVIEW

**Comment by tsenart** [comment]
@njuettner: It's idiomatic go to compare strings to the empty strings instead of using the builtin `len`. Regarding the use of the  `fmt.Sprintf` function, I can get behind that, but it'd be even better to generate the URL string from a `net/url.URL` struct.

**Comment by tsenart** [comment]
http://golang.org/pkg/net/url/

**Comment by njuettner** [comment]
ok :), I only saw it on line 76

 		if len(masters) == 0 {
 			return sj, errors.New("no master")
 		}

yes you're right it makes more sense to use url.URL

**Comment by tsenart** [comment]
The compiler optimizes `s == ""` to `len(s) == 0`.

File: records/generator.go [hunk 1]
```diff
@@ -105,9 +106,13 @@ func (rg *RecordGenerator) findMaster(leader string, masters []string) (state.St
 // Loads state.json from mesos master
 func (rg *RecordGenerator) loadFromMaster(ip string, port string) (sj state.State) {
 	// REFACTOR: state.json security
-	url := "http://" + ip + ":" + port + "/master/state.json"
-
-	req, err := http.NewRequest("GET", url, nil)
+	master := fmt.Sprintf("%s:%s", ip, port)
```
Comment (tsenart) [inline records/generator.go]: You have `net.JoinHostPort` for this exact use case. We should use that instead.

File: records/generator.go [hunk 2]
```diff
@@ -105,9 +106,13 @@ func (rg *RecordGenerator) findMaster(leader string, masters []string) (state.St
...
+	u := &url.URL{
```
Comment (tsenart) [inline records/generator.go]: No need for heap assignment. `s/&url/url/`

**Comment by tsenart** [comment]
After addressing my comments, please rebase all commits into one.

**Pull request closed without merging**